src/storage/src/hummock/compactor/mod.rs

```diff
@@ -424,40 +566,77 @@ pub fn start_compactor(
-                        event: Some(RequestEvent::ReportTask(ReportTask {
-                            task_id: compact_task.task_id,
-                            task_status: compact_task.task_status.into(),
-                            sorted_output_ssts: compact_task
-                                .sorted_output_ssts
-                                .iter()
-                                .map(|sst| sst.into())
-                                .collect(),
-                            table_stats_change: to_prost_table_stats_map(table_stats),
-                            object_timestamps: object_timestamps
-                                .iter()
-                                .map(|(object_id, timestamp)| (object_id.inner(), *timestamp))
-                                .collect(),
-                        })),
-                        create_at: SystemTime::now()
-                            .duration_since(std::time::UNIX_EPOCH)
-                            .expect("Clock may have gone backwards")
-                            .as_millis() as u64,
-                    }) {
-                        let task_id = compact_task.task_id;
-                        tracing::warn!(error = %e.as_report(), "Failed to report task {task_id:?}");
-                    }
-                }
+                                if (max_task_parallelism
+                                    - running_task_parallelism.load(Ordering::SeqCst))
+                                    < parallelism
+                                {
+                                    tracing::warn!(
+                                        "Not enough core parallelism to serve the iceberg compaction task{} task_parallelism {} running_task_parallelism {} max_task_parallelism {}",
+                                        task_id,
+                                        parallelism,
+                                        max_task_parallelism,
+                                        running_task_parallelism.load(Ordering::Relaxed),
+                                    );
+                                    continue 'consume_stream;
+                                }
 
-                match event {
-                    Some(Ok(SubscribeCompactionEventResponse { event, create_at })) => {
+                                running_task_parallelism
+                                    .fetch_add(parallelism, Ordering::SeqCst);
+                                executor.spawn(async move {
+                                    let (tx, rx) = tokio::sync::oneshot::channel();
+                                    shutdown.lock().unwrap().insert(task_id, tx);
+
+                                    icebert_runner.compact_iceberg(
+                                        rx,
+                                        parallelism as usize,
+                                    )
+                                    .await;
```

Comment:
I think we should run expire snapshot here as well, otherwise, there will be more and more snapshots.